src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -767,10 +783,14 @@ else if (TOKENS.equals(propertyName)
      */
     private static String getSinceVersionFromJavadoc(DetailNode javadoc) {
         final DetailNode sinceJavadocTag = getSinceJavadocTag(javadoc);
-        final DetailNode description = JavadocUtil.findFirstToken(sinceJavadocTag,
-                JavadocTokenTypes.DESCRIPTION);
-        final DetailNode text = JavadocUtil.findFirstToken(description, JavadocTokenTypes.TEXT);
-        return text.getText();
+        String result = null;
+        if (sinceJavadocTag != null) {
+            final DetailNode description = JavadocUtil.findFirstToken(sinceJavadocTag,
+                    JavadocTokenTypes.DESCRIPTION);
+            final DetailNode text = JavadocUtil.findFirstToken(description, JavadocTokenTypes.TEXT);
+            result = text.getText();
+        }
+        return result;
```

Comment:
We can avoid mutable data (`result`) with map of Optional:
```suggestion
    return Optional.ofNullable(sinceJavadocTag)
            .map(t -> JavadocUtil.findFirstToken(t, JavadocTokenTypes.DESCRIPTION))
            .map(d -> JavadocUtil.findFirstToken(d, JavadocTokenTypes.TEXT))
            .map(DetailNode::getText)
            .orElse(null);
```

Reply:
done